contracts/CommunityTokenDeployer.sol

```diff
@@ -0,0 +1,225 @@
+import { IAddressRegistry } from "./interfaces/IAddressRegistry.sol";
+
+/**
+ * @title CommunityTokenDeployer contract
+ * @author 0x-r4bbit
+ *
+ * This contract serves as a deployment process for Status community owners
+ * to deploy access control token contracts on behalf of their Status community.
+ * The contract keep a reference to token factories that are used for deploying tokens.
+ * The contract deploys the two token contracts `OwnerToken` and `MasterToken` those factories.
+ * The contract maintains a registry which keeps track of `OwnerToken` contract
+ * addresses per community.
+ *
+ * Only one deployment per community can be done.
+ * Status community owners have to provide an EIP712 hash signature that was
+ * created using their community's private key to successfully execute a deployment.
+ *
+ * @notice This contract is used by Status community owners to deploy
+ * community access control token contracts.
+ * @notice This contract maintains a registry that tracks contract addresses
+ * and community addresses
+ * @dev This contract will be deployed by Status, making Status the owner
+ * of the contract.
+ * @dev A contract address for a `CommunityTokenRegistry` contract has to be provided
+ * to create this contract.
+ * @dev A contract address for a `CommunityOwnerTokenFactory` contract has to be provided
+ * to create this contract.
+ * @dev A contract address for a `CommunityMasterTokenFactory` contract has to be provided
+ * to create this contract.
+ * @dev The `CommunityTokenRegistry` address can be changed by the owner of this contract.
+ * @dev The `CommunityOwnerTokenFactory` address can be changed by the owner of this contract.
+ * @dev The `CommunityMasterTokenFactory` address can be changed by the owner of this contract.
+ */
+contract CommunityTokenDeployer is EIP712("CommunityTokenDeployer", "1"), Ownable {
+    using ECDSA for bytes32;
+
+    error CommunityTokenDeployer_InvalidDeploymentRegistryAddress();
+    error CommunityTokenDeployer_InvalidTokenFactoryAddress();
+    error CommunityTokenDeployer_EqualFactoryAddresses();
+    error CommunityTokenDeployer_AlreadyDeployed();
+    error CommunityTokenDeployer_InvalidSignerKeyOrCommunityAddress();
+    error CommunityTokenDeployer_InvalidTokenMetadata();
+    error CommunityTokenDeployer_InvalidDeployerAddress();
+    error CommunityTokenDeployer_InvalidDeploymentSignature();
+
+    event OwnerTokenFactoryAddressChange(address indexed);
+    event MasterTokenFactoryAddressChange(address indexed);
+    event DeploymentRegistryAddressChange(address indexed);
+
+    /// @dev Needed to avoid "Stack too deep" error.
+    struct TokenConfig {
+        string name;
+        string symbol;
+        string baseURI;
+    }
+
+    /// @dev Used to verify signatures.
+    struct DeploymentSignature {
+        address signer;
+        address deployer;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
```

Comment:
@gravityblast I have removed the `deadline` field here.

@endulab has correctly pointed out that, given that the deployer creates the signature on behalf of the community via the community's private key, she can also set a deadline that far in the future, which renders the deadline useless.

Unless only the community controls the signature creation there's very little benefit in having it.